test/test_variables.py

```diff
@@ -108,6 +108,51 @@ def check_request_uri(req_uri):
     check_request_uri('/4%2A')
     check_request_uri('/9?q#a')
 
+    # rewrite set $request_uri before proxy
+
+    assert 'success' in client.conf(
+        {
+            "listeners": {
+                "*:8080": {"pass": "routes/a"},
+                "[::1]:8081": {"pass": "routes/b"},
+            },
+            "routes": {
+                "a": [
+                    {
+                        "action": {
+                            "rewrite": "/foo",
+                            "proxy": "http://[::1]:8081",
+                        }
+                    }
+                ],
+                "b": [
+                    {
+                        "action": {
+                            "rewrite": "/bar",
+                            "return": 200,
+                        }
+                    }
+                ],
+            },
+            "access_log": {
+                "path": f'{temp_dir}/access.log',
+                "format": "$remote_addr $uri $request_uri",
+            },
+        }
+    )
+
+    assert search_in_file(r'::1', 'access.log') is None
+
+    assert client.get(url='/blah%2Fblah?a=b')['status'] == 200
```

Comment:
Looks good.
But I can't find this type of test for normal proxy. It's useful to add such uri test without "rewrite", for example:
```
"action": {
    "proxy": "http://[::1]:8081",
}

assert client.get(url='/blah%2Fblah?a=b')['status'] == 200

# search /blah%2Fblah?a=b
assert (
        wait_for_record(fr'...', 'access.log') is not None
    ), 'req 8081 (proxy)'
```